### PR TITLE
Implement bigtable::RowSet

### LINKED ISSUE #31: Write down design doc for Row Range and Row Set wrappers
Write down a design document for row ranges and row sets.  Document the implementation steps as issues in this repository.

### PR DESCRIPTION
**Pull request by coryan**
This fixes #31.  Implement a wrapper for `::google::bigtable::v2::RowSet`.

### CODE REVIEW

**Comment by mbrukman** [comment]
What's `v3` in `::google::bigtable::v3::RowSet`?

**Comment by coryan** [comment]
A typo.  Fixed.

**Comment by DanielMahu** [comment]
Can we have a function that prunes out row keys less than or equal to a string?

I plan to use it like so: https://github.com/DanielMahu/google-cloud-cpp/commit/c887afd055b39db65cc7a26447fbbd31ff356223

**Comment by coryan** [comment]
We do not have such a function **yet**.

File: bigtable/client/row_set.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    AppendAll(std::forward<Arg&&>(a)...);
+  }
+
+  /// Add @p range to the set.
+  void Append(RowRange range) {
+    *row_set_.add_row_ranges() = range.as_proto_move();
+  }
+
+  /// Add @p row_key to the set.
+  void Append(std::string row_key) {
```
Comment (DanielMahu) [inline bigtable/client/row_set.h]: `Append(std::string)` and `Append(char const*)` are both made redundant by `Append(absl::string_view)`, any reason to keep them?

File: bigtable/client/row_set.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    *row_set_.add_row_keys() = std::move(row_key);
+  }
+
+  /// Add @p row_key to the set.
+  void Append(absl::string_view row_key) {
+    *row_set_.add_row_keys() = static_cast<std::string>(row_key);
+  }
+
+  /// Add @p row_key to the set.
+  void Append(char const* row_key) { Append(absl::string_view(row_key)); }
```
Comment (DanielMahu) [inline bigtable/client/row_set.h]: This can also be removed together with `Append(std::string)`.

File: bigtable/client/row_set.cc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/row_set.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace btproto = ::google::bigtable::v2;
```
Comment (DanielMahu) [inline bigtable/client/row_set.cc]: This file is empty

Comment (coryan) [reply bigtable/client/row_set.cc]: Removed.

File: bigtable/client/row_set.h (hunk 1 above)
Comment (coryan) [reply bigtable/client/row_set.h]: AFAIK, you cannot move from `absl::string_view` into a `std::string` so you must always copy.  The two overloads allow you to eliminate copies when possible.  If you think I am optimizing too early, happy to remove.  Also reordered the functions to explain why.

File: bigtable/client/row_set.h (hunk 2 above)
Comment (coryan) [reply bigtable/client/row_set.h]: See above.

**Comment by coryan** [comment]
If @DanielMahu (or anybody else) thinks that the overloads are premature optimization I am happy to send a second PR to remove them.

**Pull request merged**